server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -61,15 +74,65 @@ func (w *Worker) Work(ctx workflow.Context) {
 
 		msg := w.Queue.Pop()
 
-		ctx := workflow.WithValue(ctx, context.SHAKey, msg.Revision)
-		ctx = workflow.WithValue(ctx, context.DeploymentIDKey, msg.ID)
+		ctx := workflow.WithValue(ctx, internalContext.SHAKey, msg.Revision)
+		ctx = workflow.WithValue(ctx, internalContext.DeploymentIDKey, msg.ID)
+
+		// This should only happen on startup
+		// If we fail to fetch the latest deployment, log and exit the workflow
+		if w.LatestDeployment == nil {
+			w.LatestDeployment, err = w.fetchLatestDeployment(ctx, msg)
+			if err != nil {
+				logger.Error(ctx, fmt.Sprint("Unable to fetch latest deployment, worker is shutting down", err.Error()))
+				return
+			}
```

Comment:
This isn't possible right now as you haven't configured any retry cap so this will attempt to fetch the latest deployment forever.  This is probably fine for now, but I'd remove the log because I'm not sure if it's necessary since you can see the error in the UI.